titus-server-master/src/main/java/com/netflix/titus/master/agent/service/cache/DefaultAgentCache.java

```diff
@@ -348,7 +354,7 @@ private void storeEagerly(String instanceGroupId) {
     }
 
     private void storeEagerlyWithRemoveFlag(AgentInstanceGroup instanceGroup) {
-        if (!isTaggedToRemove(instanceGroup)) {
+        if (instanceGroup != null && !isTaggedToRemove(instanceGroup)) {
```

Comment:
Is this check necessary: `instanceGroup != null`?

Reply:
technically no but just defensive programming